File: Dec2023/Day-9-Dec-19-2023/2706-buy-two-chocolates/submission.py
class Solution(object):
    def buyChoco(self, prices, money):
        """
        :type prices: List[int]
        :type money: int
        :rtype: int
        """
        min1 = max(prices) 
        min2 = max(prices)
        for price in prices:
            if price < min1:
                min2 = min1
                min1 = price
            elif price < min2:
                min2 = price

        if (money-(min1+min2)) >= 0:
            return money-(min1+min2)
        else:
            return money

File: 0001-1000/python/122-best-time-to-buy-sell-stock-2/submission.scala
object Solution {
    def maxProfit(prices: Array[Int]): Int = {
        // check if it is lower or not
        // check if current > lower in prev
        // if low => update lower = current
        var maxProfit = 0
        var buyPrice = prices(0)
        for( i <- 0 to (prices.length-1)){
            // if current price is lower than previous buyPrice, make it buyPrice
            // maxProfit = Math.max(maxProfit, prices(i)-buyPrice)
            if((i>0)&&(prices(i)>prices(i-1))){
                maxProfit += prices(i) - prices(i-1)
            }
        }
        return(maxProfit)
    }
}
